### DISCUSSION TITLE
Use Pretty Blocks on CMS pages

### DISCUSSION THREAD

**Discussion by A2P2R** [PrestaSafe/prettyblocks#142] [General]
Hello, I want to use Pretty blocks to create my CMS pages. 
However, I can only add blocks in common for all CMS pages.
I would like to be able to add different blocks for each CMS page with Pretty Blocks. How can I do this? 

Thanks a lot!

**Comment by PrestaSafe** [comment]
Hi, you can use prettyblocks on every CMS pages, you need to passe the CMS id in the zone, and it's done !

**Reply by A2P2R** [reply]
hello,
thank you !

 if I understand correctly: I have to add a hook per cms id? 
do you have a tutorial to clarify what you mean?

thanks

**Reply by PrestaSafe** [reply]
Hi, no absolutely not :) 
Juste put a dynamic zone in: `themes/your_theme/templates/cms/page.tpl` 
for exemple: 
```
{block name='cms_content'}
    {$cms.content nofilter}
    {prettyblocks_zone zone_name="CMS-{$cms.id}"}
  {/block}
  ```

And go in prettyblocks, select a cms page in your shop (you need to stay in PrettyBlocks) 
and you will see your zone in the zone selector: 
<img width="353" alt="Capture d’écran 2023-09-19 à 13 30 05" src="https://github.com/PrestaSafe/prettyblocks/assets/4788787/ee6167ac-82a9-4025-9f24-d5022a863947">
This zone will by dynamic, if your are on the cms ID: 1 the zone will be `CMS-1 ` in the cms ID: 2 => `CMS-2`

😀

**Reply by A2P2R** [reply]
@PrestaSafe 
it's beautiful, thank you very much!

 is it possible to do this for other types of content on a prestashop site? 

example: add content with pretty block on certain categories or products (and not on a template)

thank you

**Reply by PrestaSafe** [reply]
@A2P2R you're welcome. 
Yes of course you can put content anywhere with this tips. 
Just put the good id ;)

**Comment by aomaxime** [comment]
Bonjour, j'ai ajouté `{prettyblocks_zone zone_name="CMS-{$cms.id}"}` sur ma page `themes/your_theme/templates/cms/page.tpl` mais la zone ne m'est pas proposé 

![image](https://github.com/PrestaSafe/prettyblocks/assets/148320838/7a59f62c-3270-4a18-bd2e-5f7e27ec9df1)

**Reply by PrestaSafe** [reply]
Bonjour, 
Vous êtes vous rendu sur une page cms depuis prettyblocks ? Avez vous vidé le cache de prestashop ?

**Reply by PrestaSafe** [reply]
je confirme que c'est ok pour moi:    
 `{prettyblocks_zone zone_name="CMS-{$cms.id}"}` 
dans 
`themes/montheme/templates/cms/page.tpl` 
<img width="380" alt="Capture d’écran 2023-10-18 à 09 41 35" src="https://github.com/PrestaSafe/prettyblocks/assets/4788787/7d363c56-2324-4b91-90fc-308f46109e41">

**Reply by aomaxime** [reply]
Bonjour, je n'avais pas compris la subtilité de devoir naviguer jusqu'à la page depuis l'interface d'édition. En l’occurrence ma page n'était pas accessible depuis la home donc aucun moyen d'y accéder. Peut-être une amélioration à apporter :)

**Reply by PrestaSafe** [reply]
Oui et non, car si il y a beaucoup de zones, ça risque de devenir compliqué. 
Le plus simple reste de naviguer sur une page. Je vais voir pour que ce soit plus simple, via un menu déroulant ou autre. 
Cordialement

**Reply by aomaxime** [reply]
Je comprends la problématique, merci pour votre réponse rapide en tout cas :)
